debian/rules

```diff
@@ -6,15 +6,16 @@
 # assume we only have one package
 PACKAGE_NAME:=`dh_listpackages`
 
-override_dh_systemd_enable:
-	dh_systemd_enable --name=matrix-synapse
-
-override_dh_installinit:
-	dh_installinit --name=matrix-synapse
+override_dh_installsystemd:
+	dh_installsystemd --name=matrix-synapse
 
 # we don't really want to strip the symbols from our object files.
 override_dh_strip:
 
+# we also don't need to optimise the size of the debug information, this also
+# avoids a crash
```

Comment:
```suggestion
# we also don't need to optimise the size of the debug information. This also
# avoids a crash.
```

Tell me more about this crash? If `dwz` is segfaulting that sounds like a debian bug: do you have a bug reference?

Reply:
> we also don't need to optimise the size of the debug information

why not? it sounds like a good thing.

Reply:
I didn't bother looking for a bug reference, because before it crashed (I don't remember on which library) the changes were either minuscule (in the tens of bytes) or dwz discarded its work because it actually increased the size.

Reply:
ok, can you please share the details of the crash for posterity

Reply:
```
dwz: Allocatable section in debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/PIL/_imaging.cpython-39-x86_64-linux-gnu.so after non-allocatable ones
dwz: Allocatable section in debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/PIL/_imagingcms.cpython-39-x86_64-linux-gnu.so after non-allocatable ones
dwz: Allocatable section in debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/PIL/_imagingft.cpython-39-x86_64-linux-gnu.so after non-allocatable ones
dwz: Allocatable section in debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/PIL/_webp.cpython-39-x86_64-linux-gnu.so after non-allocatable ones
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libXau-00ec42fe.so.6.0.0: .debug_info section not present
dwz: Allocatable section in debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libfreetype-a029e222.so.6.18.1 after non-allocatable ones
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libharfbuzz-851aa43c.so.0.30200.0: .debug_info section not present
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libjpeg-62ed1500.so.62.3.0: .debug_info section not present
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/liblcms2-5ee890d7.so.2.0.13: .debug_info section not present
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/liblzma-d540a118.so.5.2.5: .debug_info section not present
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libopenjp2-430a98fc.so.2.4.0: .debug_info section not present
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libpng16-213e245f.so.16.37.0: .debug_info section not present
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libtiff-8e99fb9e.so.5.7.0: .debug_info section not present
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libwebp-8efe125f.so.7.1.3: .debug_info section not present
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libwebpdemux-016472e8.so.2.0.9: .debug_info section not present
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libwebpmux-5c00cf3e.so.3.0.8: .debug_info section not present
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libxcb-1122e22b.so.1.1.0: .debug_info section not present
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libz-dd453c56.so.1.2.11: .debug_info section not present
dwz: Allocatable section in debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/_cffi_backend.cpython-39-x86_64-linux-gnu.so after non-allocatable ones
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/cffi.libs/libffi-9c61262e.so.8.1.0: .debug_info section not present
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/ijson.libs/libyajl-36341dea.so.2.1.0: .debug_info section not present
dwz: Allocatable section in debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/ijson/backends/_yajl2.cpython-39-x86_64-linux-gnu.so after non-allocatable ones
dwz: debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/cryptography/hazmat/bindings/_openssl.abi3.so: DWARF compression not beneficial - old size 622627 new size 632277
dh_dwz: error: dwz -mdebian/matrix-synapse-py3/usr/lib/debug/.dwz/x86_64-linux-gnu/matrix-synapse-py3.debug -M/usr/lib/debug/.dwz/x86_64-linux-gnu/matrix-synapse-py3.debug -- debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/PIL/_imaging.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/PIL/_imagingcms.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/PIL/_\
imagingft.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/PIL/_imagingmath.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/PIL/_imagingmorph.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/PIL/_imagingtk.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/PIL/_web\
p.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libXau-00ec42fe.so.6.0.0 debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libfreetype-a029e222.so.6.18.1 debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libharfbuzz-851aa43c.so.0.30200.0 debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libjpeg-62ed1500.so.6\
2.3.0 debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/liblcms2-5ee890d7.so.2.0.13 debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/liblzma-d540a118.so.5.2.5 debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libopenjp2-430a98fc.so.2.4.0 debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libpng16-213e245f.so.16.37.0 debian/matrix-synapse-py3/\
opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libtiff-8e99fb9e.so.5.7.0 debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libwebp-8efe125f.so.7.1.3 debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libwebpdemux-016472e8.so.2.0.9 debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libwebpmux-5c00cf3e.so.3.0.8 debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/pyt\
hon3.9/site-packages/Pillow.libs/libxcb-1122e22b.so.1.1.0 debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/Pillow.libs/libz-dd453c56.so.1.2.11 debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/_cffi_backend.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/bcrypt/_bcrypt.abi3.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/cffi.libs/libffi-9c61262e.so.8\
.1.0 debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/cryptography/hazmat/bindings/_openssl.abi3.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/cryptography/hazmat/bindings/_rust.abi3.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/frozendict/_frozendict.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/hiredis/hiredis.cpython-39-x86_64-linux-gnu.\
so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/ijson.libs/libyajl-36341dea.so.2.1.0 debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/ijson/backends/_yajl2.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/lxml/_elementpath.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/lxml/builder.cpython-39-x86_64-linux-gnu.so debian\
/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/lxml/etree.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/lxml/html/clean.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/lxml/html/diff.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/lxml/objectify.cpython-39-x86_64-linux-gnu.so debian/matrix-syn\
apse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/lxml/sax.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/markupsafe/_speedups.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/msgpack/_cmsgpack.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/nacl/_sodium.abi3.so debian/matrix-synapse-py3/opt/venvs/matrix-syna\
pse/lib/python3.9/site-packages/psycopg2/_psycopg.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/pvectorc.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/simplejson/_speedups.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/systemd/_daemon.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/\
lib/python3.9/site-packages/systemd/_journal.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/systemd/_reader.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/systemd/id128.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/systemd/login.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/pyt\
hon3.9/site-packages/thrift/protocol/fastbinary.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/tornado/speedups.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/yaml/_yaml.cpython-39-x86_64-linux-gnu.so debian/matrix-synapse-py3/opt/venvs/matrix-synapse/lib/python3.9/site-packages/zope/interface/_zope_interface_coptimizations.cpython-39-x86_64-linux-gnu.so returned exit code 1
make: *** [debian/rules:58: binary] Error 1
```

Reply:
ok, that doesn't sound like a "crash" to me; it's just a regular error.

can you tell which of the lines are errors and which are just warnings/diagnostics? I suspect it's just the _openssl.abi3 one that is an error?

Reply:
That did look a bit differently, when I tried this initially, but this is what I got while trying to reproduce this just now. So your mileage may vary, but yes the only error here is the one for openssl.

Reply:
I've tried a few rebuilds in the background and I can't seem to get `dwz` to *not* error out with exit code 1. I excluded the above openssl module and tried the erroring cmdline itself, but even in the best case it prints a long list of no ops and returns 1. If you want I can update the comment (including that this should be revisited in the future), but I don't think `override_dh_dwz` will do anything worthwhile for this package for now.

Reply:
Right, thanks for taking the time to investigate this.

It seems like the "Allocatable section after non-allocatable ones" messages are errors, and the libraries it is complaining about do indeed have allocatable sections after non-allocatable ones, which, as far as I can tell from a brief search on the internet is considered wrong and probably a bug in the toolchain used to build the libraries in question.

Since the libraries in question are being pulled from pypi, and emprically working fine, I agree it's not worth pursuing this further.

So if you can update the comment to say something along the lines of "dwz errors out with messages like 'Allocatable section after non-allocatable ones', and produces marginal results anyway", that would be great.

Reply:
I see. Interesting, that this is such a common error.

I pushed a commit updating the comment.